src/Data/DataFrame/IO/CSV.hs

```diff
@@ -129,11 +131,21 @@ inferValueType s = let
 -- | Reads rows from the handle and stores values in mutable vectors.
 fillColumns :: Int -> Char -> VM.IOVector Column -> VM.IOVector [(Int, T.Text)] -> Handle -> IO ()
 fillColumns n c mutableCols nullIndices handle = do
+    input <- newIORef (mempty :: C.ByteString)
     forM_ [1..n] $ \i -> do
         isEOF <- hIsEOF handle
-        unless isEOF $ do
-            row <- parseSep c <$> C.hGetLine handle
-            zipWithM_ (writeValue mutableCols nullIndices i) [0..] row
+        input' <- readIORef input
+        unless (isEOF && input' == mempty) $ do
+              parseWith (C.hGet handle 512) (parseRow c) input' >>= \case
+                Fail unconsumed ctx er -> do
+                  erpos <- hTell handle
+                  fail $ "Failed to parse CSV file around " <> show erpos <> " byte; due: "
+                    <> show er <> "; context: " <> show ctx
+                Partial c -> do
+                  fail "Partial handler is called"
```

Comment:
What happens here for rows longer than 512? Shouldn't we just keep accumulating until there's enough data in the ioref?

Reply:
It tested with 31 bytes instead of 512. So buffer was definitely shorter than a row.
My observation Partial branch is never called.
I guess it is for really partial files only when you need to download content from network eg.